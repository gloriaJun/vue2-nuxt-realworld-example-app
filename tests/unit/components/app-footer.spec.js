import { shallowMount } from '@vue/test-utils';
import AppFooter from '@/components/app-footer.vue';

const siteName = 'siteName';

describe('AppFooter.vue', () => {

  const build = () => {
    const wrapper = shallowMount(AppFooter, {
      stubs: ['nuxt-link'],
      propsData: { siteName: siteName },
    });

    return {
      wrapper,
    }
  }

  it('should match the snapshot', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('check logo link', () => {
    const { wrapper } = build();

    wrapper.find('.logo-font').trigger('click');
    expect(window.location.href).toBe(`http://${window.location.host}/`);
  });

  it('check logo name', () => {
    const { wrapper } = build();
    expect(wrapper.find('.logo-font').text()).toBe(siteName);
  });
})
