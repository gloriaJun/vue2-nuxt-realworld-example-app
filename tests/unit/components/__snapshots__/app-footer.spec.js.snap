// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.vue should match the snapshot 1`] = `
<footer>
  <div class="container">
    <nuxt-link-stub to="/" class="logo-font">
      siteName
    </nuxt-link-stub> <span class="attribution">
      An interactive learning project from
      <a href="https://thinkster.io">
        Thinkster
      </a>.
      Code &amp; design licensed under MIT.
    </span>
  </div>
</footer>
`;
